server/src/MessageHandler.ts

```diff
@@ -26,56 +63,103 @@ function getWordAtPosition(document: TextDocument, position: Position): string {
   return currentLine.slice(beginning, end + position.character)
 }
 
-export async function handleDefinitionRequest(
+export class MyBlock {
+  type: string
+  start: Position
+  end: Position
+  name: string
+
+  constructor(type: string, start: Position, end: Position, name: string) {
+    this.type = type
+    this.start = start
+    this.end = end
+    this.name = name
+  }
+}
+
+function getBlockAtPosition(
+  line: number,
+  document: TextDocument,
+): MyBlock | void {
+  if (document) {
+    let blockType = ''
+    let blockName = ''
+    let blockStart: Position = Position.create(0, 0)
+    let blockEnd: Position = Position.create(0, 0)
+    // get block beginning
+    for (let i = line; i >= 0; i--) {
+      const currentLine = getCurrentLine(document, i).trim()
+      if (currentLine.includes('{')) {
+        // position is inside a block
+        const index = currentLine.indexOf(' ')
+        blockType = ~index ? currentLine.slice(0, index) : currentLine
+        blockName = currentLine
+          .substring(blockType.length, currentLine.length - 2)
```

Comment:
To reduce mental overhead while reasoning about the code, I would either fully use `substr` and `substring` everywhere or use `slice` everywhere. (as `slice` and `substring` have [slightly different behavior](https://stackoverflow.com/questions/2243824/what-is-the-difference-between-string-slice-and-string-substring) I personally prefer `slice` for everything, but that's a matter of taste.